File: app/src/main/java/mk/webfactory/template/feature/common/ui/BaseActivity.java
package mk.webfactory.template.feature.common.ui;

import android.os.Bundle;
import android.support.annotation.Nullable;
import androidx.core.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import dagger.android.AndroidInjection;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.HasSupportFragmentInjector;
import javax.inject.Inject;
import mk.webfactory.template.App;

public abstract class BaseActivity extends AppCompatActivity implements HasSupportFragmentInjector {

  @Inject DispatchingAndroidInjector<Fragment> supportFragmentInjector;

  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
    AndroidInjection.inject(this);
    super.onCreate(savedInstanceState);
  }

  @Override protected void onResume() {
    super.onResume();
    App.CRASH_REPORT.setCurrentPage(getClass().getCanonicalName());
  }

  @Override public AndroidInjector<Fragment> supportFragmentInjector() {
    return supportFragmentInjector;
  }
}

File: app/src/main/java/mk/webfactory/template/feature/common/ui/BaseActivity.kt
package mk.webfactory.template.feature.common.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import dagger.android.AndroidInjection
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasAndroidInjector
import mk.webfactory.template.App.Companion.CRASH_REPORT
import javax.inject.Inject

abstract class BaseActivity : AppCompatActivity(), HasAndroidInjector {

    @Inject
    lateinit var androidInjector: DispatchingAndroidInjector<Any>

    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState?: Bundle())
    }

    override fun onResume() {
        super.onResume()
        CRASH_REPORT.setCurrentPage(javaClass.canonicalName!!)
    }

    override fun androidInjector(): AndroidInjector<Any> {
        return androidInjector
    }
}